File: app/src/main/java/com/neome/java/api/meta/base/dto/StudioFieldPickUser.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.MetaIdField;
import com.neome.java.api.meta.base.Types.MetaIdRole;
import com.neome.java.api.meta.base.Types.MetaIdVar;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class StudioFieldPickUser extends StudioFieldEditable
{
  @Nullable
  public MetaIdVar dataSourceVarId;

  @Nullable
  public MetaIdRole defaultValue;

  @Nullable
  public MetaIdField defaultValueFieldId;

  @Nullable
  public StudioDtoPluginApi pluginApi;

  @Nullable
  public MetaIdField pluginErrorFieldId;

  @Nullable
  public MetaIdVar pluginInputMappingVarId;

  @Nullable
  public MetaIdRole[] roleIdDataSource;

  @Nullable
  public Boolean showAsDropdown;

  @Nullable
  public MetaIdField showAsDropdownFieldId;

  @Nullable
  public MetaIdVar showAsDropdownVarId;
}

File: app/src/main/java/com/neome/api/meta/base/dto/StudioFieldPickUser.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdRole
import com.neome.api.meta.base.Types.MetaIdVar

interface StudioFieldPickUser : StudioFieldEditable {
    val dataSourceVarId: MetaIdVar?
    val defaultValue: MetaIdRole?
    val defaultValueFieldId: MetaIdField?
    val pluginApi: StudioDtoPluginApi?
    val pluginErrorFieldId: MetaIdField?
    val pluginInputMappingVarId: MetaIdVar?
    val roleIdDataSource: List<MetaIdRole>?
    val showAsDropdown: Boolean?
    val showAsDropdownFieldId: MetaIdField?
    val showAsDropdownVarId: MetaIdVar?
}
